presto-thrift-connector/src/test/java/com/facebook/presto/connector/thrift/clientproviders/MockConnectedThriftServiceProvider.java

```diff
@@ -0,0 +1,155 @@
+
+public class MockConnectedThriftServiceProvider
+        implements ConnectedThriftServiceProvider
+{
+    private static final PrestoThriftServiceException RETRIABLE_EXCEPTION = new PrestoThriftServiceException("Mocking a retryable thrift failure", true);
+    private static final HostAddress DUMMY_HOST_ADDRESS = fromParts("test_host", 1234);
+    private static final HostLocationHandle DUMMY_HOST_LOCATION_HANDLE = new HostLocationHandle()
+    {
+        @Override
+        public HostAddress getHostAddress()
+        {
+            return DUMMY_HOST_ADDRESS;
+        }
+
+        @Override
+        public void reportFailure(Exception e) {}
+    };
+
+    private final AtomicInteger pendingFailuresToGenerate = new AtomicInteger();
+    private final AtomicInteger openServiceCount = new AtomicInteger();
+
+    public MockConnectedThriftServiceProvider() {}
+
+    public void setPendingFailuresToGenerate(int scheduledFailures)
+    {
+        this.pendingFailuresToGenerate.set(scheduledFailures);
+    }
+
+    public int getOpenServiceCount()
+    {
+        return openServiceCount.get();
+    }
+
+    @Override
+    public ConnectedThriftService anyHostClient()
+    {
+        openServiceCount.getAndIncrement();
+        return new MockConnectedThriftService(pendingFailuresToGenerate.getAndDecrement() > 0);
+    }
+
+    @Override
+    public ConnectedThriftService selectedHostClient(List<HostAddress> hosts)
+    {
+        openServiceCount.getAndIncrement();
+        return new MockConnectedThriftService(pendingFailuresToGenerate.getAndDecrement() > 0);
+    }
+
+    private final class MockConnectedThriftService
+            implements ConnectedThriftService
+    {
+        private final boolean failedService;
+
+        MockConnectedThriftService(boolean failedService)
+        {
+            this.failedService = failedService;
+        }
+
+        void failIfConfigured()
+        {
+            if (failedService) {
+                throw RETRIABLE_EXCEPTION;
+            }
+        }
+
+        @Override
+        public HostLocationHandle getHostLocationHandle()
+        {
+            return DUMMY_HOST_LOCATION_HANDLE;
+        }
+
+        @Override
+        public List<String> listSchemaNames()
+                throws PrestoThriftServiceException
+        {
+            failIfConfigured();
+            return ImmutableList.of();
+        }
+
+        @Override
+        public List<PrestoThriftSchemaTableName> listTables(PrestoThriftNullableSchemaName schemaNameOrNull)
+                throws PrestoThriftServiceException
+        {
+            failIfConfigured();
+            return ImmutableList.of();
+        }
+
+        @Override
+        public PrestoThriftNullableTableMetadata getTableMetadata(PrestoThriftSchemaTableName schemaTableName)
+                throws PrestoThriftServiceException
+        {
+            throw new UnsupportedOperationException("getTableMetadata is not supported in MockConnectedThriftService");
```

Comment:
You don't need a message like this in test implementations

Reply:
Need to throw something and have some words in the message. I think it should be fine.

Reply:
You can keep the text, but remove class and method names from the message.